test/suite/test_compact10.py

```diff
@@ -183,19 +182,25 @@ def test_compact10_incr_backup(self):
 
         # Every time compaction makes progress, perform an incremental backup.
         bytes_recovered = 0
-        while self.get_files_compacted(uris) < self.num_tables:
+        files_compacted = 0
+        while self.get_files_compacted(uris) + files_compacted < self.num_tables:
             new_bytes_recovered = self.get_bytes_recovered()
             if new_bytes_recovered != bytes_recovered:
                 # Update the incremental backup ID from the parent class.
                 self.bkup_id += 1
-                self.take_incr_backup(self.backup_dir_2)
-                bytes_recovered = new_bytes_recovered
-
-        assert self.get_files_compacted(uris) == self.num_tables
 
-        # Compare the backups.
-        for uri in uris:
-            self.compare_backups(uri, self.backup_dir_1, self.backup_dir_2)
+                self.take_incr_backup(self.backup_incr)
+                self.take_full_backup(self.backup_full)
```

Comment:
Why are we taking a full backup? As described at the start of the test, we want to perform only incremental backups as compaction is running.

Reply:
fixed up now! [4cdad97](https://github.com/wiredtiger/wiredtiger/pull/9977/commits/4cdad97f9da9cc86715e95394d6b77009a7b2343)